File: src/WebhookBuilder.java
package net.glitchtechs.starapi;

import java.io.OutputStream;
import java.net.HttpURLConnection;
import java.net.URL;
import java.nio.charset.StandardCharsets;

public class WebhookBuilder {

    private final String webhookUrl;
    private String message;

    private WebhookBuilder(String webhookUrl) {
        this.webhookUrl = webhookUrl;
    }

    public static WebhookBuilder create(String webhookUrl) {
        return new WebhookBuilder(webhookUrl);
    }

    public WebhookBuilder addMessage(String message) {
        this.message = message;
        return this;
    }

    public void build() {
        try {
            URL url = new URL(webhookUrl);
            HttpURLConnection connection = (HttpURLConnection) url.openConnection();

            connection.setRequestMethod("POST");
            connection.setRequestProperty("Content-Type", "application/json");
            connection.setDoOutput(true);

            String payload = "{\"content\":\"" + message + "\"}";

            try (OutputStream outputStream = connection.getOutputStream()) {
                byte[] input = payload.getBytes(StandardCharsets.UTF_8);
                outputStream.write(input, 0, input.length);
            }

            int responseCode = connection.getResponseCode();
            System.out.println("HTTP Response Code: " + responseCode);

            connection.disconnect();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: Kotlin/src/WebhookBuilder.kt
package net.glitchtechs.starapi

import java.net.HttpURLConnection
import java.net.URL
import java.nio.charset.StandardCharsets

class WebhookBuilder private constructor(private val webhookUrl: String) {
    private var message: String? = null

    companion object {
        fun create(webhookUrl: String): WebhookBuilder {
            return WebhookBuilder(webhookUrl)
        }
    }

    fun addMessage(message: String): WebhookBuilder {
        this.message = message
        return this
    }

    fun build() {
        try {
            val url = URL(webhookUrl)
            val connection = url.openConnection() as HttpURLConnection

            connection.requestMethod = "POST"
            connection.setRequestProperty("Content-Type", "application/json")
            connection.doOutput = true

            val payload = "{\"content\":\"$message\"}"

            connection.outputStream.use { outputStream ->
                val input = payload.toByteArray(StandardCharsets.UTF_8)
                outputStream.write(input, 0, input.size)
            }

            val responseCode = connection.responseCode
            println("HTTP Response Code: $responseCode")

            connection.disconnect()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}
